### DISCUSSION TITLE
Expressions in Docker URI

### DISCUSSION THREAD

**Discussion by pheel** [community/community#26382] [Actions]
<p>I’m trying to pass an ${{ expression }} inside a Docker URI in my workflow, as such:</p>
<pre><code>**# ./github/workflows/main.yml**  

steps:
...
   - name: launch api
     uses: 'docker://${{ steps.previous_step.outputs.image }}'
</code></pre>
<p>Result is the error:</p>
<pre><code>Unrecognized named-value: 'steps'.
</code></pre>
<p>As an alternative, I’ve tried:</p>
<pre><code>**# ./github/workflows/main.yml**

steps:
   ...
   - name: launch api
     uses: ./.github/actions/forward-tags/
with:
image: ${{ steps.previous_step.outputs.image }

**# ./github/actions/forward-tags/action.yml** name: 'Forward tags'
inputs:
  image:
    required: true
runs:
  using: 'docker'
  image: 'docker://${{ inputs.image }}'
</code></pre>
<p>Result is the error:</p>
<pre><code>Unrecognized named-value: 'inputs'.
</code></pre>
<p>Has anyone succesfully used expressions this way?</p>
<p>Thanks!</p>

**Comment by pheel** [comment]
<p>I was able to run it using:</p>
<pre><code>steps:
...
   - name: launch api
     uses: actions/docker/cli@master
     with:
       args: 'run ${{ steps.previous_step.outputs.image }}'
</code></pre>
<p>However, it would be great to support uses: ‘docker://${{ steps.previous_step.outputs.image }}’ directly and be able to set Docker environment variables as properties of this step, etc.</p>

**Comment by mosteo** [comment]
<p>I had this exact problem without finding a solution. I will try the workaround but I would still like to know if this is a deliberate design decision, and the reasons in that case.</p>

**Comment by mosteo** [comment]
<p>Actually my use case is the proposed workaround from the OP: having the action take a parameter which is the dockerhub image name.</p>

**Comment by chrispat** [comment, marked as answer]
<p>We currently do not support expressions in the uses tag in order to to know which repos we need to pull down as we are processing the job.  We also have a feature that limits the usage of third party actions and if actions are defined as an expression we would not be able to block that before the job starts on the runner.</p>

**Comment by mosteo** [comment]
<p>Hello <a class="mention" href="/chrispat">@chrispat</a> , thanks for the clarification.</p>
<p>In my use case the action is statically given and the expression is in the <code>image:</code> parameter of <code>runs:</code>, entirely within the action. So perhaps this is a different issue to the restrictions you explain in your reply?</p>

**Comment by elprans** [comment]
<p>Lack of support for expressions in the “uses” tag imposes quite an inconvenience for docker-based actions.  Would it be possible to specify a path to a Dockerfile using an expression in the <code>with</code> stanza?</p>

**Comment by BobbyMcWho** [comment]
<p>I also would like to be able to use the</p>
<pre><code>name: 'Forward tags'
inputs:
  image:
    required: true
runs:
  using: 'docker'
  image: 'docker://${{ inputs.image }}'
</code></pre>
<p>syntax.</p>

**Comment by mosteo** [comment]
<p>I created <a href="https://github.com/marketplace/actions/run-within-docker" rel="noopener nofollow ugc">this action</a> as a workaround for this use case. I’m using it since some time successfully.</p>
